### PR TITLE
Implement new trade offer logic

### PR DESCRIPTION
**Pull request by i509VCB**
And some whitespace for the builder.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/item/merchant/SpongeTradeOfferBuilder.java [hunk 1]
```diff
@@ -53,9 +54,12 @@
     private int useCount;
     private int maxUses;
     private boolean allowsExperience;
+    private int merchantExperienceGranted;
+    private double priceGrowthMultiplier;
+    private int demandBonus;
 
     public SpongeTradeOfferBuilder() {
-        super(TradeOffer.class, 1);
+        super(TradeOffer.class, 2);
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/item/merchant/SpongeTradeOfferBuilder.java]: What's the difference here?

File: src/main/java/org/spongepowered/common/item/merchant/SpongeTradeOfferBuilder.java [hunk 2]
```diff
@@ -68,6 +72,7 @@ public SpongeTradeOfferBuilder() {
 
     @Override
     public TradeOffer.Builder secondBuyingItem(final ItemStack item) {
+        Preconditions.checkNotNull(item, "Second buying item cannot be null");
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/item/merchant/SpongeTradeOfferBuilder.java]: Can the second item not be null? I thought we allowed it to be null implicitly?

**Review by gabizou — APPROVED** [review]
Null check aside, I'd be willing to make the second buying item as a null setter as well, but nitpicks are nitpicks.

File: src/main/java/org/spongepowered/common/item/merchant/SpongeTradeOfferBuilder.java (hunk 2 above)
Comment (i509VCB) [reply src/main/java/org/spongepowered/common/item/merchant/SpongeTradeOfferBuilder.java]: The item can be empty.

**Pull request merged**